webapp/commonComponents/expression/expressionVariables.js

```diff
@@ -52,30 +65,34 @@ const getChildDefVariables = (survey, nodeDefContext, nodeDefCurrent, mode, lang
       }
 
       if (mode === Expression.modes.sql) {
-        return getSqlVariables(childDef, lang)
+        return getSqlVariables(childDef, nodeDefCurrent, i18n)
       }
 
       if (mode === Expression.modes.json) {
-        return getJsVariables(childDef, lang)
+        return getJsVariables(childDef, nodeDefCurrent, i18n)
       }
 
       return null
     }),
     R.flatten,
-    R.reject(R.isNil),
+    R.reject(R.isNil)
   )(survey)
 
-export const getVariables = (survey, nodeDefContext, nodeDefCurrent, mode, preferredLang) => {
-  const lang = Survey.getLanguage(preferredLang)(Survey.getSurveyInfo(survey))
+export const getVariables = (survey, nodeDefContext, nodeDefCurrent, mode, i18n) => {
   const surveyWithDependencies = Survey.buildAndAssocDependencyGraph(survey)
 
   const variables = []
-  Survey.visitAncestorsAndSelf(nodeDefContext, nodeDef => {
+  Survey.visitAncestorsAndSelf(nodeDefContext, (nodeDef) => {
     if (!NodeDef.isVirtual(nodeDef) || !NodeDef.isEqual(nodeDefContext)(nodeDef)) {
-      const childVariables = getChildDefVariables(surveyWithDependencies, nodeDef, nodeDefCurrent, mode, lang)
+      const childVariables = getChildDefVariables(surveyWithDependencies, nodeDef, nodeDefCurrent, mode, i18n)
       variables.push(...childVariables)
     }
   })(surveyWithDependencies)
 
+  if (nodeDefCurrent) {
+    // Move current node def variable to the first position
+    const nodeDefCurrentUuid = NodeDef.getUuid(nodeDefCurrent)
+    variables.sort((varA, varB) => (varB.uuid === nodeDefCurrentUuid) - (varA.uuid === nodeDefCurrentUuid))
```

Comment:
besides the nodeDefCurrent to top, should we sort them alphabetically as well?

Reply:
ok